File: c++/learn_c++_with_robot_project/components/src/Gripper.cpp
#include "components/Gripper.hpp"
#include <iostream>
#include <string>

Gripper::Gripper(int id, const std::string& name)
    : RobotComponent(id, name),
    m_is_open(true),
    m_force_newton(0.0) {}

void Gripper::open() {
    m_is_open = true;
}

void Gripper::close() {
    m_is_open = false;
}

void Gripper::setForce(double force) {
    m_force_newton = force;
}

double Gripper::getForce() const {
    return m_force_newton;
}

void Gripper::printStatus() const {
    RobotComponent::printStatus();
    
    std::string state = m_is_open ? "Open" : "Close";
    std::cout << "State: " << state << "\n" << "Force: " << m_force_newton << " N\n";
}

File: python/learn_python_with_robot_project/components/gripper.py
from .component import RobotComponent

class Gripper(RobotComponent):
    def __init__(self, component_id, name):
        super().__init__(component_id, name)
        self.__is_open = True
        self.__force_newton = 0.0
    
    def open(self):
        self.__is_open = True

    def close(self):
        self.__is_open = False

    def set_force(self, force):
        self.__force_newton = force

    def get_force(self):
        return self.__force_newton
    
    def print_status(self):
        super().print_status()
        state = lambda state : "Open" if self.__is_open else "Close"
        print(f"State: {state}\nForce: {self.__force_newton} N\n")